aspnetcore/tutorials/choose-web-ui.md

```diff
@@ -0,0 +1,193 @@
+
+Blazor components have the flexibility of being hosted either using Blazor Server or Blazor WebAssembly.
+
+Blazor WebAssembly is a [single-page app (SPA) framework](/dotnet/architecture/modern-web-apps-azure/choose-between-traditional-web-and-single-page-apps) for building interactive client-side web apps that run in the browser on a [WebAssembly](https://webassembly.org/) based .NET runtime (*Blazor WebAssembly*).
+
+Benefits of Blazor WebAssembly:
+
+* Uses open web standards without plugins or recompiling code into other languages.
+* Works with all modern web browsers, including mobile browsers.
+* WebAssembly code can access the full functionality of the browser via JavaScript, called *JavaScript interoperability*.
+
+To get started with ASP.NET Core Blazor WebAssembly, see [Tooling for ASP.NET Core Blazor](xref:blazor/tooling). For an overview of ASP.NET Core Blazor WebAssembly, its architecture and benefits, see [Overview of Blazor WebAssembly](xref:blazor/index#blazor-webassembly).
+
+#### ASP.NET Core Blazor Server
+
+ASP.NET Core Blazor Server has the advantage of direct access to server resources, while providing model for building rich, interactive, and composable user interfaces in C#. Blazor Server components are UI elements such as a page, dialog, or data entry form, that handle client-side web UI interactions. The client-side web UI interactions are handled over a websocket connection, and can span multiple connections to increase speed. Blazor components keep track of the state of the client-side DOM and efficiently apply updates to only the components that need it.
+
+Blazor Server benefits:
+
+* Share app logic across server and client..
+* Flexibility of direct access to server resources. For example:
+  * Database.
+  * Secrets, such as access key-values used for API calls to Azure storage.
+* Uses a token-based security model, rather than using cookies. For example, cookies may not be enabled on the browser.
+
+To get started with ASP.NET Core Blazor Server, see [Get started with Blazor](https://dotnet.microsoft.com/learn/aspnet/blazor-tutorial/intro). For an overview of ASP.NET Core Blazor Server, its architecture and benefits, see [Introduction to ASP.NET Core Blazor](xref:blazor/index).
+
+### ASP.NET Core MVC plus hosted Blazor WebAssembly
+
+MVC and Blazor are both part of the ASP.NET Core framework and are designed to be used together. Razor components can be integrated into Razor Pages and MVC apps in a hosted Blazor WebAssembly solution. When the page or view is rendered, components can be pre-rendered at the same time.
+
+Choose ASP.NET Core MVC plus Blazor WebAssembly when these benefits best fit your Web UI app needs:
+
+* Prerendering:
+  * Executes Blazor components on the server and renders them into a page or view.
+	* Improves the perceived load time of the app while interactivity is being set up.
+* Add islands of interactivity to existing views (pages), using the component tag helper.
```

Comment:
I think 🤔 essential for this TH remark (i.e., drop the comma) ... and all THs are proper nouns ...

```suggestion
* Add islands of interactivity to existing views (pages) using the Component Tag Helper.
```

... or for sure if "with" is used ...

```suggestion
* Add islands of interactivity to existing views (pages) with the Component Tag Helper.
```

If you want to cross link it, it's at ....

```
[Component Tag Helper](xref:mvc/views/tag-helpers/builtin-th/component-tag-helper)
```